src/System.Drawing.Primitives/tests/SizeTests.netcoreapp.cs

```diff
@@ -0,0 +1,132 @@
+        }
+
+
+        [Theory]
+        [InlineData(1000, 2400.933f)]
+        [InlineData(int.MaxValue, 0.0f)]
+        [InlineData(int.MaxValue, 1.0f)]
+        [InlineData(int.MaxValue, -1.0f)]
+        [InlineData(int.MaxValue, float.MaxValue)]
+        [InlineData(int.MinValue, 0.0f)]
+        [InlineData(int.MinValue, 1.0f)]
+        [InlineData(int.MinValue, -1.0f)]
+        [InlineData(int.MinValue, float.MinValue)]
+        [InlineData(int.MaxValue, float.MinValue)]
+        [InlineData(int.MinValue, float.MaxValue)]
+        public void MultiplicationTestSizeFloat(int value1, float value2)
+        {
+            Size sz1 = new Size(value1, value1);
+            SizeF mulExpected;
+
+            unchecked
+            {
+                mulExpected = new SizeF(value1 * value2, value1 * value2);
+            }
+
+            Assert.Equal(mulExpected, sz1 * value2);
+            Assert.Equal(mulExpected, value2 * sz1);
+        }
+
+        [Fact]
+        public void DivideByZeroChecks()
+        {
+            Size size = new Size(100, 100);
+            Assert.Throws<DivideByZeroException>(() => size / 0);
+
+            float invDiv = 1 / 0.0f;
+            SizeF expectedSizeF = new SizeF(float.PositiveInfinity, float.PositiveInfinity);
+            Assert.Equal(expectedSizeF, size * invDiv);
+        }
+
+        [Theory]
+        [InlineData(0, 1)]
+        [InlineData(1, 1)]
+        [InlineData(-1, 1)]
+        [InlineData(1, -1)]
+        [InlineData(-1, -1)]
+        [InlineData(int.MaxValue, int.MaxValue)]
+        [InlineData(int.MaxValue, int.MinValue)]
+        [InlineData(int.MinValue, int.MaxValue)]
+        [InlineData(int.MinValue, int.MinValue)]
+        [InlineData(int.MaxValue, 1)]
+        [InlineData(int.MinValue, 1)]
+        [InlineData(int.MaxValue, -1)]
+        public void DivideTestSizeInt(int value1, int value2)
+        {
+            Size size = new Size(value1, value1);
+            Size expected;
+
+            int invDiv = 1 / value2;
+            expected = new Size(value1 * invDiv, value1 * invDiv);
+
+            Assert.Equal(expected, size / value2);
+        }
+
+        [Theory]
+        [InlineData(0, 1.0f)]
+        [InlineData(1, 1.0f)]
+        [InlineData(-1, 1.0f)]
+        [InlineData(1, -1.0f)]
+        [InlineData(-1, -1.0f)]
+        [InlineData(int.MaxValue, float.MaxValue)]
+        [InlineData(int.MaxValue, float.MinValue)]
+        [InlineData(int.MinValue, float.MaxValue)]
+        [InlineData(int.MinValue, float.MinValue)]
+        [InlineData(int.MaxValue, 1.0f)]
+        [InlineData(int.MinValue, 1.0f)]
+        [InlineData(int.MaxValue, -1.0f)]
+        [InlineData(int.MinValue, -1.0f)]
+        public void DivideTestSizeFloat(int value1, float value2)
+        {
+            SizeF size = new SizeF(value1, value1);
+            SizeF expected;
+
+            float invDiv = 1.0f / value2;
+
+            expected = new SizeF(value1 * invDiv, value1 * invDiv);
```

Comment:
@WinCPP, the purpose of doing the two separate operations (multiply by reciprocal in the product code and actual division in the test code) is to validate that we didn't mess up the operator by performing some optimization or refactoring.

We want to multiply by reciprocal in the product code because it is faster.

We want to divide in the test to validate that our refactoring/optimization didn't break anything.

In this case, we end up with a slightly different result for some cases. However, the minor variations in the final result is generally considered acceptable (especially in code that deals with general purpose graphics). The general allowed variance is 6-9 significant digits, so provided our final result (even with the optimization) is within that range, we are still good.